doc/classes/Array.xml

```diff
@@ -58,29 +48,32 @@
 			<param index="2" name="class_name" type="StringName" />
 			<param index="3" name="script" type="Variant" />
 			<description>
-				Creates a typed array from the [param base] array. All arguments are required.
-				- [param type] is the built-in type as a [enum Variant.Type] constant, for example [constant TYPE_INT].
-				- [param class_name] is the [b]native[/b] class name, for example [Node]. If [param type] is not [constant TYPE_OBJECT], must be an empty string.
-				- [param script] is the associated script. Must be a [Script] instance or [code]null[/code].
-				Examples:
+				Creates a typed array from the [param base] array. A typed array can only contain elements of the given type, or that inherit from the given class, as described by this constructor's parameters:
+				- [param type] is the built-in [Variant] type, as one the [enum Variant.Type] constants.
+				- [param class_name] is the built-in class name (see [method Object.get_class]).
+				- [param script] is the associated script. It must be a [Script] instance or [code]null[/code].
+				If [param type] is not [constant TYPE_OBJECT], [param class_name] must be an empty [StringName] and [param script] must be [code]null[/code].
 				[codeblock]
-				class_name MyNode
+				class_name Sword
 				extends Node
 
-				class MyClass:
+				class Stats:
 				    pass
 
 				func _ready():
-				    var a = Array([], TYPE_INT, &amp;"", null) # Array[int]
-				    var b = Array([], TYPE_OBJECT, &amp;"Node", null) # Array[Node]
-				    var c = Array([], TYPE_OBJECT, &amp;"Node", MyNode) # Array[MyNode]
-				    var d = Array([], TYPE_OBJECT, &amp;"RefCounted", MyClass) # Array[MyClass]
+				    var a = Array([], TYPE_INT, "", null)               # Array[int]
+				    var b = Array([], TYPE_OBJECT, "Node", null)        # Array[Node]
+				    var c = Array([], TYPE_OBJECT, "Node", Sword)       # Array[Sword]
+				    var d = Array([], TYPE_OBJECT, "RefCounted", Stats) # Array[Stats]
 				[/codeblock]
-				[b]Note:[/b] This constructor can be useful if you want to create a typed array on the fly, but you are not required to use it. In GDScript you can use a temporary variable with the static type you need and then pass it:
+				The [param base] array's elements are converted when necessary. If this is not possible or [param base] is already typed, this constructor fails and returns an empty [Array].
+				In GDScript, this constructor is usually not necessary, as it is possible to create a typed array through static typing:
 				[codeblock]
-				func _ready():
-				    var a: Array[int] = []
-				    some_func(a)
+				var numbers: Array[float] = []
+				var children: Array[Node] = get_children()
```

Comment:
This is not a very good example, since here the array is not created, but an already existing one is assigned. I would suggest the following:

```suggestion
				var children: Array[Node] = [$Node1, $Node2, $Node3]
```

Reply:
Will do, with name changes to also hint towards the objects inheriting from "Node" that matters here.